services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.repository.ts

```diff
@@ -65,6 +65,30 @@ export class ProgramFinancialServiceProviderConfigurationRepository extends Repo
     return response;
   }
 
+  public async getUsernamePasswordPropertiesByVoucherId(
+    intersolveVoucherId: number,
+  ): Promise<UsernamePasswordInterface> {
+    const programFspConfig = await this.baseRepository
+      .createQueryBuilder('configuration')
+      .leftJoin('configuration.transactions', 'transactions')
+      .leftJoin('transactions.registration', 'registration')
+      .leftJoin('registration.images', 'images')
+      .leftJoin('images.voucher', 'voucher')
+      .where('voucher.id = :intersolveVoucherId', {
+        intersolveVoucherId,
+      })
+      .andWhere('voucher.payment = transactions.payment') // REFACTOR: this filter is needed, as it is not taken care of by the joins above. Better refactor the entity relations here.
+      .getOne(); // the above can still return multiple transaction (steps/attempts) for the same voucher, but they will always have the same fspConfig, so take one
```

Comment:
1. You could get the latest transaction. That should be the right one right? `innerjoin('transactions.latestTransaction)` or `sort('transactions.create')`
2. May also add a comment at this method that this should be refactored when we pick up the Ah voucher refactor and than critically look at module responsibility
3. I am not sure about this. But I have some memory that a getOne can be relatively slow with many left join. As the object mapping of typeorm can be slow. We ran into this once on the GET program endpoint, for which we than created the `getProgramWithManyAttributes.js` type. So maybe you can use .getRawOne combined with .select(). Or a regular findOne() with a select...